Xamarin.PropertyEditing.Mac/Controls/RectangleEditorControl.cs

```diff
@@ -14,29 +14,28 @@ protected RectangleEditorControl (IHostResourceProvider hostResources)
 			: base (hostResources)
 		{
 			// TODO localize
-			XLabel.Frame = new CGRect (34, 28, 25, 22);
-			XLabel.Font = NSFont.FromFontName (DefaultFontName, DefaultDescriptionLabelFontSize); // TODO: Washed-out color following specs
 			XLabel.StringValue = "X";
 
 			XEditor.Frame = new CGRect (0, 46, 90, 20);
-
-			YLabel.Frame = new CGRect (166, 28, 25, 22);
-			YLabel.Font = NSFont.FromFontName (DefaultFontName, DefaultDescriptionLabelFontSize); // TODO: Washed-out color following specs
+			
 			YLabel.StringValue = "Y";
 
 			YEditor.Frame = new CGRect (132, 46, 90, 20);
-
-			WidthLabel.Frame = new CGRect (20, -5, 50, 22);
-			WidthLabel.Font = NSFont.FromFontName (DefaultFontName, DefaultDescriptionLabelFontSize); // TODO: Washed-out color following specs
+			
 			WidthLabel.StringValue = "WIDTH";
 
 			WidthEditor.Frame = new CGRect (0, 13, 90, 20);
-
-			HeightLabel.Frame = new CGRect (150, -5, 50, 22);
-			HeightLabel.Font = NSFont.FromFontName (DefaultFontName, DefaultDescriptionLabelFontSize); // TODO: Washed-out color following specs
+			
 			HeightLabel.StringValue = "HEIGHT";
 
 			HeightEditor.Frame = new CGRect (132, 13, 90, 20);
+
+			this.AddConstraints (new[] {
+				NSLayoutConstraint.Create (XLabel, NSLayoutAttribute.Left, NSLayoutRelation.Equal, XEditor, NSLayoutAttribute.CenterX, 1f, -10f),
+				NSLayoutConstraint.Create (YLabel, NSLayoutAttribute.Left, NSLayoutRelation.Equal, YEditor, NSLayoutAttribute.CenterX, 1f, -10f),
+				NSLayoutConstraint.Create (WidthLabel, NSLayoutAttribute.Left, NSLayoutRelation.Equal, WidthEditor, NSLayoutAttribute.CenterX, 1f, -25f),
+				NSLayoutConstraint.Create (HeightLabel, NSLayoutAttribute.Left, NSLayoutRelation.Equal, HeightEditor, NSLayoutAttribute.CenterX, 1f, -25f),
```

Comment:
You can specify `CenterX` on the `HeightLabel` so you don't need to do any manual adjustment that will assuredly break when the labels are allowed to be changed.

`NSLayoutConstraint.Create (HeightLabel, NSLayoutAttribute.CenterX, NSLayoutRelation.Equal, HeightEditor, NSLayoutAttribute.CenterX, 1f, 0),`

Reply:
No, because it will centre on the whole control, including spinners, which we don't want it to do, we just want it centred on the editing part of the control. At least that was the initial design from @vancura . If that has changed, then I'll change it to be centre on everything.

Reply:
You could include a manual offset for the spinners themselves, a negative offset of half the width of the spinners.

Reply:
The Labels were hard coded as `X`, `Y`, `WIDTH`, `HEIGHT` etc because we had a discussion about how they wouldn't change. Otherwise I would have put them in for localisation resources. We discussed this with Alan like a year or possibly more ago.

Reply:
We discussed them not being localized which is not the same as not being changed, they are a property name and therefore subject to the platform. We haven't implemented the functionality yet, but they will absolutely have the option of changing in the future. Beyond that, your manual offset is still subject to the font and font size. Is there another technical concern with doing it this way?

Reply:
I think calculating the offsets, with font info will incur an overhead, but I've made the change. We'll find out if that is likely or not. Just awaiting Vaclav's approval that it looks fine for him.

Reply:
You're right, that would incur an overhead. Was there an issue with my suggested solution?

Reply:
@ermau  Do you mean this...
> You could include a manual offset for the spinners themselves, a negative offset of half the width of the spinners.

?

Reply:
> Do you mean this?

That and
`NSLayoutConstraint.Create (HeightLabel, NSLayoutAttribute.CenterX, NSLayoutRelation.Equal, HeightEditor, NSLayoutAttribute.CenterX, 1f, 0),`

Reply:
`NSLayoutConstraint.Create (HeightLabel, NSLayoutAttribute.CenterX, NSLayoutRelation.Equal, HeightEditor, NSLayoutAttribute.CenterX, 1f, 0),` will not take into account the width of the actual text in the label, it will position it exactly in the centre of the label. Adding the spinner offsets, unfortunately isn't enough for it to position the text correctly. 

As allowing custom labels in on the road-map, doing the width calculations is the best way to ensure we handle things correctly, regardless of font and the text users might use. Please take a look at the calculations. There might be another way to minimise the calculations overheads. It doesn't look massive right now, but might be worth another pair of eyes.

Reply:
> will not take into account the width of the actual text in the label, it will position it exactly in the centre of the label.

The label already sizes to text, how would the center of the label not also be the center of the text?
<img width="792" alt="image" src="https://user-images.githubusercontent.com/156582/53050627-c7362100-3467-11e9-9bb2-59923dd286d9.png">

I checked out your branch and tested this change to the thickness editor (we can actually skip the spinner offset by constraint directly to the textbox):
```csharp
this.AddConstraints (new[] {
	NSLayoutConstraint.Create (XLabel, NSLayoutAttribute.CenterX, NSLayoutRelation.Equal, XEditor.Subviews[0], NSLayoutAttribute.CenterX, 1f, 0),
	NSLayoutConstraint.Create (YLabel, NSLayoutAttribute.CenterX, NSLayoutRelation.Equal, YEditor.Subviews[0], NSLayoutAttribute.CenterX, 1f, 0),
	NSLayoutConstraint.Create (WidthLabel, NSLayoutAttribute.CenterX, NSLayoutRelation.Equal, WidthEditor.Subviews[0], NSLayoutAttribute.CenterX, 1f, 0),
	NSLayoutConstraint.Create (HeightLabel, NSLayoutAttribute.CenterX, NSLayoutRelation.Equal, HeightEditor.Subviews[0], NSLayoutAttribute.CenterX, 1f, 0),
});
```

<img width="310" alt="image" src="https://user-images.githubusercontent.com/156582/53050765-18deab80-3468-11e9-90ed-09c12b2f30f8.png">

As you can measure, the thickness labels are centered perfectly (give or take 1px because the widths aren't even) on the textbox above it.

Reply:
I hadn't thought of using `SubViews[0]`. That works really well and simplifies the code across the board and no calcs on our part.

Reply:
Done.